egs/madcat_zh/v1/local/create_line_image_from_page_image.py

```diff
@@ -0,0 +1,587 @@
+    y_dash_4 = (y4 - center_y) * cos(rotation_angle_in_rad) + (x4 - center_x) * sin(rotation_angle_in_rad) + center_y
+    return x_dash_1, y_dash_1, x_dash_2, y_dash_2, x_dash_3, y_dash_3, x_dash_4, y_dash_4
+
+
+def pad_image(image):
+    """ Pads the image around the border. It help in getting
+        bounding boxes that are slightly outside the page boundary.
+    Args:
+        image: page image.
+
+    Returns:
+        image: page image
+    """
+
+    #Temporary change
+    #height = image.shape[0]
+    #im_pad = np.concatenate((255 * np.ones((height, width_padding),
+    #                                     dtype=int), image), axis=1)
+    #im_pad1 = np.concatenate((im_pad, 255 * np.ones((height, width_padding),
+    #                                              dtype=int)), axis=1)
+
+    #width = im_pad1.shape[1]
+
+    #im_pad2 = np.concatenate((255 * np.ones((height_padding, width),
+    #                                      dtype=int), im_pad1), axis=0)
+    #im_pad3 = np.concatenate((im_pad2, 255 * np.ones((height_padding, width),
+    #                                               dtype=int)), axis=0)
+
+    padded_image = Image.new('RGB', (image.size[0] + padding, image.size[1] + padding), "white")
+    padded_image.paste(im=image, box=(offset, offset))
+    return padded_image
+
+
+def update_minimum_bounding_box_input(bounding_box_input):
+    """ Updates the word bounding box corner points.
+    Args:
+        points [(float, float)]: points, a list or tuple of 2D coordinates.
+                                 ideally should be more than 2 points
+    Returns:
+        points [(float, float)]: points, a list or tuple of 2D coordinates
+    """
+
+    updated_minimum_bounding_box_input = []
+    for point in bounding_box_input:
+        x, y = point
+        new_x = x + offset
+        new_y = y + offset
+        word_coordinate = (new_x, new_y)
+        updated_minimum_bounding_box_input.append(word_coordinate)
+
+    return updated_minimum_bounding_box_input
+
+
+def set_line_image_data(image, line_id, image_file_name):
+    """ Flips a given line image and saves it. Line image file name 
+        is formed by appending the line id at the end page image name.
+    Args:
+        image: line image, non flipped
+        line_id (string): id of the line image.
+        image_file_name(string): name of the page image.
+
+    Returns:
+    """
+
+    base_name = os.path.splitext(os.path.basename(image_file_name))[0]
+    image_file_name_wo_tif, b = image_file_name.split('.tif')
+    line_id = '_' + line_id.zfill(4)
+    line_image_file_name = base_name + line_id + '.tif'
+    image_path = os.path.join(output_directory, line_image_file_name)
+    #Temporary change
+    #imgray_rev_arr = np.fliplr(image)
+    #im = img_as_uint(imgray_rev_arr)
+    #imsave(image_path, im)
+    imgray = image.convert('L')
+    imgray_rev_arr = np.fliplr(imgray)
```

Comment:
Sorry, previously did it at the wrong location. This line flips the image, please remove this line.

Reply:
Ah yes, I saw that error. I've fixed it in a commit but never pushed it to github.